src/pages/index.js

```diff
@@ -6,46 +6,83 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import ProfilePicEvt from "../components/ProfilePicEvt.js";
 import {
-  initialCards,
   config,
   editProfileButton,
+  editProfilePictureButton,
   addCardButton,
+  profilePicture,
 } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+let gallery;
 //Add Card Popup Form
 const addCardPopupForm = new PopupWithForm("#add-card-modal", {
   handleSubmitEvent: (inputValues) => {
-    const newCard = {
-      name: inputValues.name,
-      link: inputValues.bio,
-    };
-    const cardElement = createCard(newCard);
-    gallery.addItem(`prepend`, cardElement);
+    addCardPopupForm.renderLoading(true);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        const cardElement = createCard(inputValues);
+        gallery.addItem(`prepend`, cardElement);
+        addCardPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        addCardPopupForm.renderLoading(false);
+      });
     addCardPopupForm.form.reset();
   },
 });
 // Edit Popup Form
 const editPopupForm = new PopupWithForm("#edit-profile-modal", {
   handleSubmitEvent: (inputValues) => {
-    userInfo.setUserInfo({ name: inputValues.name, job: inputValues.bio });
+    editPopupForm.renderLoading(true);
+    api
+      .updateUserInfo(inputValues)
+      .then((inputValues) => {
+        userInfo.setUserInfo({
+          name: inputValues.name,
+          job: inputValues.about,
+        });
+        editPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        editPopupForm.renderLoading(false);
```

Comment:
`....renderLoading(false)` should be in `finally`, not to duplicate it in `then` and `catch`

**Please, correct it everywhere in the code**